.github/workflows/main.yml

```
@@ -11,8 +11,29 @@ concurrency:
   cancel-in-progress: ${{ !contains(github.ref, 'refs/heads/main') }}
 
 jobs:
+  check-skip-merge-queue:
+    name: Check if pull request can skip merge queue
+    runs-on: ubuntu-latest
+    outputs:
+      skip-merge-queue: ${{ steps.check-skip-merge-queue.outputs.up-to-date }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v6
+        if: github.event_name == 'merge_group'
+      - name: Check pull request merge queue status
+        id: check-skip-merge-queue
+        if: github.event_name == 'merge_group'
+        uses: ./.github/actions/check-merge-queue
+        with:
+          head-ref: ${{ github.event.merge_group.head_ref }}
+          base-ref: main
+          github-token: ${{ secrets.GITHUB_TOKEN }}
```

Nit: Given that this is a secret, we can pass it as a secret instead:

```suggestion
        secrets:
          github-token: ${{ secrets.GITHUB_TOKEN }}
```

We don't want GitHub to treat this as non-sensitive, and log it in the workflow logs or something.

This would require changes to the check-merge-queue action as well, to change it from an input to a secret

---

This is one of the confusing differences between actions and workflows, but actions don't accept secrets, and using inputs is actually the recommended way to provide secrets to actions. GitHub will still treat them as sensitive values.

For reference, from [the docs](https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#jobsjob_idsecrets):

> When a job is used to call a reusable workflow, you can use `secrets` to provide a map of secrets that are passed to the called workflow.

Notice how there's no `jobs.<job_id>.steps[*].secrets`, and `jobs.<job_id>.uses` is only valid for workflows.